Cabal/Distribution/Simple/Utils.hs

```diff
@@ -1380,21 +1381,25 @@ withTempDirectoryEx _verbosity opts targetDir template f = withFrozenCallStack $
 --     ASCII-representable. Since Cabal-3.0 the file is assumed to be
 --     UTF-8 encoded.
 rewriteFileEx :: Verbosity -> FilePath -> String -> IO ()
-rewriteFileEx verbosity path newContent =
+rewriteFileEx verbosity path =
+  rewriteFileExBS verbosity path . toUTF8LBS
+
+-- | Same as `rewriteFileEx` but for 'ByteString's.
+rewriteFileExBS :: Verbosity -> FilePath -> BS.ByteString -> IO ()
```

Comment:
This is lazy bytestring, isn't it?

We unfortunately do `import qualified Data.ByteString.Lazy as BS` in this module, but the name of function should have `LBS` prefix.

Reply:
Fixed